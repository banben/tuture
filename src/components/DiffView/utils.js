function computeOldLineNumber(change) {
  if (change.isInsert) {
    return -1;
  }

  return change.isNormal ? change.oldLineNumber : change.lineNumber;
}

function computeNewLineNumber(change) {
  if (change.isDelete) {
    return -1;
  }

  return change.isNormal ? change.newLineNumber : change.lineNumber;
}

function getChangeKey(change) {
  if (!change) {
    throw new Error('change is not provided');
  }

  const {
    isNormal,
    isInsert,
    lineNumber,
    oldLineNumber,
  } = change;

  if (isNormal) {
    return 'N' + oldLineNumber;
  }

  const prefix = isInsert ? 'I' : 'D';
  return prefix + lineNumber;
}

export {
  computeOldLineNumber,
  computeNewLineNumber,
  getChangeKey,
}
