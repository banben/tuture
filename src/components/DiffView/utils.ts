import { Change as ChangeType } from '../../types';

function computeOldLineNumber(change: ChangeType): number {
  if (change.isInsert) {
    return -1;
  }

  return change.isNormal ? change.oldLineNumber : change.lineNumber;
}

function computeNewLineNumber(change: ChangeType): number {
  if (change.isDelete) {
    return -1;
  }

  return change.isNormal ? change.newLineNumber : change.lineNumber;
}

function getChangeKey(change: ChangeType): string {
  if (!change) {
    throw new Error('change is not provided');
  }

  const {
    isNormal,
    isInsert,
    lineNumber,
    oldLineNumber,
  } = change;

  if (isNormal) {
    return 'N' + oldLineNumber;
  }

  const prefix = isInsert ? 'I' : 'D';
  return prefix + lineNumber;
}

export {
  computeOldLineNumber,
  computeNewLineNumber,
  getChangeKey,
}
