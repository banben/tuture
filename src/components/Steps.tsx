import React from 'react';
import styled from 'styled-components';

interface updateSelectFunc {
  (key: number): void;
}

interface catalogsInfo {
  name: string;
  language: string;
  maintainer: string;
  topics: string[];
}

interface catalog {
  name: string;
  commit: string;
}

interface StepsProps {
  selectKey: number;
  updateSelect: updateSelectFunc;
  catalogs: catalog[];
  catalogsInfo: catalogsInfo;
}

const TutureSteps = styled.div`
  width: 20%;
  height: 100%;
  overflow: scroll;
  border-right: 1px solid #e8e8e8;
`;

const TutureMenu = styled.ul`
  height: 100%;
  overflow: scroll;
  font-size: 14px;
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  outline: none;
  margin-bottom: 0;
  padding-left: 0;
  list-style: none;
  color: rgba(0, 0, 0, .65);
  line-height: 0;
  background: #fff;
  -webkit-transition: background .3s, width .2s;
  transition: background .3s, width .2s;
  zoom: 1;
`;

const TutureMenuItem = styled.li`
  padding: 0 16px 0 30px;
  width: calc(100% + 1px);
  font-size: 14px;
  line-height: 40px;
  height: 40px;
  margin-top: 4px;
  margin-right: 2px;
  overflow: hidden;
  text-overflow: ellipsis;
  cursor: pointer;
  margin: 0;
  display: block;
  white-space: nowrap;
  border-right: ${(props: any) => props.className === 'selected' ? '4px solid #1890ff' : 'none'};
  color: ${(props: any) => props.className === 'selected' ? '#1890ff' : '#000000'};
  background-color: ${(props: any) => props.className === 'selected' ? '#e6f7ff' : '#FFFFFF'};
`;

export default class Steps extends React.Component<StepsProps> {
  render() {
    const {
      catalogs,
      catalogsInfo,
      selectKey,
      updateSelect,
    } = this.props;

    return (
      <TutureSteps>
        <TutureMenu>
          {
            catalogs.map((item, key) => (
              <TutureMenuItem
                className={key === selectKey ? 'selected' : ''}
                onClick={() => { this.props.updateSelect(key) }}
                key={key}
              >
              {item.name}
              </TutureMenuItem>
            ))
          }
        </TutureMenu>
      </TutureSteps>
    );
  }
}
