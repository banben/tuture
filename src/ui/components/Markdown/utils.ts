const spliceStr = (str = '', typeStr: string, start: number, end: number) => {
  return str
    .slice(0, start)
    .concat(
      typeStr,
      str.slice(start, end),
      typeStr,
      str.slice(end, str.length),
    );
};

const insertStr = (str = '', typeStr: string, insertPosition: number) => {
  return str
    .slice(0, insertPosition)
    .concat(typeStr, str.slice(insertPosition, str.length));
};

export { spliceStr, insertStr };
