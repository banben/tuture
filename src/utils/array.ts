const isArray = (arr: any): boolean => {
  return Array.isArray(arr);
};

export {
  isArray,
};
