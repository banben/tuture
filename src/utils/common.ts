function handleAnchor(origin: string) {
  return origin.toLowerCase().replace(/ /g, '-');
}

function isClientOrServer() {
  return typeof window !== 'undefined' && window.document ? 'client' : 'server';
}

const reorder = (list: any[], startIndex: number, endIndex: number): any[] => {
  const result = Array.from(list);
  const [removed] = result.splice(startIndex, 1);
  result.splice(endIndex, 0, removed);

  return result;
};

const vwDesign = 1680;
const vwFontsize = 168;

function rem(px: number) {
  return px / vwFontsize;
}

export { handleAnchor, isClientOrServer, reorder, rem, vwDesign, vwFontsize };
