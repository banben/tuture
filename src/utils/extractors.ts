const extractCommits = (tuture: Tuture) => {
  const commits = tuture.steps.map((item: Step) => ({
    name: item.name,
    commit: item.commit,
  }));

  return commits;
};

const extractMetaData = (tuture: Tuture) => {
  const { name, topics, description } = tuture;
  const metadata = {
    name,
    topics,
    description,
  };

  return metadata;
};

export { extractCommits, extractMetaData };
