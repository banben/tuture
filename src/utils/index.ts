import * as obj from './object';
import * as arr from './array';

const tutureUtilities = {
  ...arr,
  ...obj,
};

export default tutureUtilities;
