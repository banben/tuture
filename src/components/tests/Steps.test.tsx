import React from 'react';
import renderer from 'react-test-renderer';

import Steps from '../Steps';
import { catalogs, catalogsInfo } from './utils/data';

test('Test Steps render successfully', () => {
  let selectKey = 0;
  const updateSelect = (key: number) => {
    selectKey = key;
  };

  const component = renderer.create(
    <Steps
      catalogs={catalogs}
      catalogsInfo={catalogsInfo}
      selectKey={selectKey}
      updateSelect={updateSelect}
    />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
