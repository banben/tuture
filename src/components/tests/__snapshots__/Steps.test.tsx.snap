// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Steps render successfully 1`] = `
<div
  className="sc-bdVaJa cSstg"
>
  <ul
    className="sc-bwzfXH cHPvXJ"
  >
    <li
      className="selected sc-htpNat iRATYX"
      onClick={[Function]}
    >
      sssssssssssssssssssss
    </li>
    <li
      className="sc-htpNat dFEOip"
      onClick={[Function]}
    >
      服务器端依赖安装
    </li>
    <li
      className="sc-htpNat dFEOip"
      onClick={[Function]}
    >
      数据库的配置与数据表的设计
    </li>
    <li
      className="sc-htpNat dFEOip"
      onClick={[Function]}
    >
      编写 User 表路由和处理文件
    </li>
    <li
      className="sc-htpNat dFEOip"
      onClick={[Function]}
    >
      编写 Topics 路由极其处理文件
    </li>
    <li
      className="sc-htpNat dFEOip"
      onClick={[Function]}
    >
      编写 Notifications 路由及其处理文件
    </li>
    <li
      className="sc-htpNat dFEOip"
      onClick={[Function]}
    >
      使用 Postman 测试结果
    </li>
    <li
      className="sc-htpNat dFEOip"
      onClick={[Function]}
    >
      初始化 Client 工程
    </li>
    <li
      className="sc-htpNat dFEOip"
      onClick={[Function]}
    >
      Eject 项目，加入必要依赖
    </li>
    <li
      className="sc-htpNat dFEOip"
      onClick={[Function]}
    >
      构建主页
    </li>
    <li
      className="sc-htpNat dFEOip"
      onClick={[Function]}
    >
      编写 Main 页面
    </li>
    <li
      className="sc-htpNat dFEOip"
      onClick={[Function]}
    >
      编写 Notifications 组件
    </li>
    <li
      className="sc-htpNat dFEOip"
      onClick={[Function]}
    >
      编写 Topic 组件
    </li>
    <li
      className="sc-htpNat dFEOip"
      onClick={[Function]}
    >
      编写 Topics 组件
    </li>
    <li
      className="sc-htpNat dFEOip"
      onClick={[Function]}
    >
      修改 App，完成整个项目
    </li>
    <li
      className="sc-htpNat dFEOip"
      onClick={[Function]}
    >
      add content for debug
    </li>
    <li
      className="sc-htpNat dFEOip"
      onClick={[Function]}
    >
      renderer: completed new features
    </li>
    <li
      className="sc-htpNat dFEOip"
      onClick={[Function]}
    >
      add switch view and hide files
    </li>
    <li
      className="sc-htpNat dFEOip"
      onClick={[Function]}
    >
      addjust stylesheet
    </li>
    <li
      className="sc-htpNat dFEOip"
      onClick={[Function]}
    >
      test for renderer
    </li>
    <li
      className="sc-htpNat dFEOip"
      onClick={[Function]}
    >
      add global tuture-renderer command
    </li>
  </ul>
</div>
`;
